### PR TITLE
Include properties of an `export =` value in import completions.

### LINKED ISSUE #13218: Import completion list for properties of `export =` value
**TypeScript Version:**  nightly (2.2.0-dev.201xxxxx)

**Code**

```ts
///<reference path="fourslash.ts"/>

// @Filename: /node_modules/modash/index.d.ts
////export = Modash;
////declare var Modash: Modash.Static;
////declare namespace Modash {
////    interface Static {
////        foo(): void;
////    }
////}

// @Filename: /a.ts
////import { /**/ } from "modash";

goTo.marker();
verify.completionListContains("foo");
```

**Expected behavior:**

Test succeeds.

**Actual behavior:**

Test fails. `Static` is completed but `foo` is not.

Original issue: DefinitelyTyped/DefinitelyTyped#13557.

### PR DESCRIPTION
**Pull request by andy-ms**
Fixes #13218

Wish there was a way to do this without exposing `resolveExternalModuleSymbol`, but I wouldn't be comfortable modifying `getExportsOfModule`.

### CODE REVIEW

File: src/services/completions.ts [hunk 1]
```diff
@@ -1322,6 +1322,12 @@ namespace ts.Completions {
             const moduleSpecifierSymbol = typeChecker.getSymbolAtLocation(importOrExportDeclaration.moduleSpecifier);
             if (moduleSpecifierSymbol) {
                 exports = typeChecker.getExportsOfModule(moduleSpecifierSymbol);
+                const exportEquals = typeChecker.resolveExternalModuleSymbol(moduleSpecifierSymbol);
+                if (exportEquals !== moduleSpecifierSymbol) {
+                    // Location doesn't matter so long as it's not an identifier.
+                    const exportEqualsType = typeChecker.getTypeOfSymbolAtLocation(exportEquals, moduleSpecifier);
+                    exports = ts.concatenate(exports, typeChecker.getPropertiesOfType(exportEqualsType));
```
Comment (mhegazy) [inline src/services/completions.ts]: why do we need to concatenate here? how can you have exports along with `export=`?

Comment (andy-ms) [inline src/services/completions.ts]: Like in the test file, you could have both `var Foo` and `namespace Foo`. `exports` (first set) is for `namespace Foo`, `getPropertiesOfType` is for `var Foo`.

Comment (mhegazy) [inline src/services/completions.ts]: aah.. i see.

**Comment by mhegazy** [comment]
I would rather we have this in one place.. like in the checker, call getExportedSymbolsFromModule, instead of having to do this in the services.

**Pull request merged**